### DISCUSSION TITLE
Email Notifications for New Entries

### DISCUSSION THREAD

**Discussion by Armaanmehta11** [supabase/supabase#28513] [Questions]
Hi everyone,

I'm currently working on implementing an email notification system that sends out a single email summarizing all new table entries added (potentially deleted/updated) each day, rather than sending an email for each individual trigger event. Is there a way to accomplish this directly within Supabase, or would I need to continue using a separate microservice as I'm doing now?

Any guidance or suggestions would be greatly appreciated!

Thanks!

**Comment by GaryAustin1** [comment]
You could have a cron task run once a day and check for updated at since last run.  The cron task function could use http/pg_net to call your email edge function or service.   One issue would be the edge case of what time the last cron ran and getting table entries from that point as it would not be down to the second.

Edit:  You could also have a trigger function write to a "queue" table.  Then do cron on the queue table and delete the entries you get when you have sent them.

**Reply by Armaanmehta11** [reply]
I have currently implemented a similar solution using a cron job to periodically pull database tables and perform diff checks against the previous state. I'm exploring whether a similar approach can be implemented directly using Supabase functions and triggers with no external scripts? I found [this article](https://medium.com/@DeolaJ/how-to-trigger-email-notifications-on-a-contact-form-with-supabase-210cf245e57e), which demonstrates how to trigger email notifications for each new entry. However, sending an email for every update would be too frequent for my needs.

**Reply by GaryAustin1** [reply]
Yes.  So I described using a cron task once a day.  You either pull all "new" entries to the table by date or use a queue table (generated by a trigger on the original table) that has all the entries from the last time cron ran.  You bundle up all those row changes and present them in the body of your email.  Use send the data for the email like that link shows or to an edge function to do further processing.
